test/forge/CarbonVortex.sol

```diff
@@ -0,0 +1,644 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import { Test } from "forge-std/Test.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+import { TestFixture } from "./TestFixture.sol";
+import { CarbonVortex } from "../../contracts/vortex/CarbonVortex.sol";
+
+import { AccessDenied, InvalidAddress, InvalidFee } from "../../contracts/utility/Utils.sol";
+import { PPM_RESOLUTION } from "../../contracts/utility/Constants.sol";
+
+import { IVoucher } from "../../contracts/voucher/interfaces/IVoucher.sol";
+import { ICarbonController } from "../../contracts/carbon/interfaces/ICarbonController.sol";
+import { ICarbonVortex } from "../../contracts/vortex/interfaces/ICarbonVortex.sol";
+import { IBancorNetwork } from "../../contracts/vortex/CarbonVortex.sol";
+
+import { Token, toIERC20, NATIVE_TOKEN } from "../../contracts/token/Token.sol";
+
+contract CarbonVortexTest is TestFixture {
+    using Address for address payable;
+
+    address private bancorNetworkV3;
+
+    uint256 private constant REWARDS_PPM_DEFAULT = 100_000;
+    uint256 private constant REWARDS_PPM_UPDATED = 110_000;
```

Comment:
Minor: we use `uint32` for all PPM values.

Reply:
Actually, not for CarbonVortex, due to a recent upgrade replacing the rewards storage struct with rewards ppm variable.